analyzer/src/main/resources/scripts/init.gradle

```diff
@@ -279,9 +279,9 @@ class AbstractDependencyTreePlugin<T> implements Plugin<T> {
                             error, warning, pomFile, null)
                 } else if (id instanceof ProjectComponentIdentifier) {
                     def dependencyProject = project.rootProject.findProject(id.projectPath)
-                    return new DependencyImpl(dependencyProject.group.toString(), dependencyProject.name,
-                            dependencyProject.version.toString(), '', '', dependencies, null, null, null,
-                            dependencyProject.projectDir.absolutePath)
+                    return new DependencyImpl(groupId: dependencyProject.group.toString(),
+                            artifactId: dependencyProject.name, version: dependencyProject.version.toString(),
+                            dependencies: dependencies, localPath: dependencyProject.projectDir.absolutePath)
```

Comment:
As there is now only one place left where the tuple constructor is used, what about using the map constructor and removing the tuple constructor annotation again?

Reply:
Yeah, I was a bit undecided about this. When all arguments are specified, using the map-constructor / named arguments seems a bit verbose. And if I removed `@TupleConstructor` from `DependencyImpl` it would seem inconsistent to keep it for the other classes, but the other classes are always called with all arguments. So I decided to keep `@TupleConstructor` to offer the best of both worlds.